src/conftest.py

```diff
@@ -0,0 +1,143 @@
+            'postgres-rw-user': config.cfg.data_storage.postgres_rw_user,
+            'postgres-rw-pw': config.cfg.data_storage.postgres_rw_pw,
+            'postgres-del-user': config.cfg.data_storage.postgres_del_user,
+            'postgres-del-pw': config.cfg.data_storage.postgres_del_pw,
+            'postgres-admin-user': config.cfg.data_storage.postgres_del_user,
+            'postgres-admin-pw': config.cfg.data_storage.postgres_del_pw,
+
+            'redis-fact-db': config.cfg.data_storage.redis_test_db,  # Note: This is unused in testing
+            'redis-test-db': config.cfg.data_storage.redis_test_db,  # Note: This is unused in production
+            'redis-host': config.cfg.data_storage.redis_host,
+            'redis-port': config.cfg.data_storage.redis_port,
+
+            'firmware-file-storage-directory': str(firmware_file_storage_directory),
+
+            'user-database': 'sqlite:////media/data/fact_auth_data/fact_users.db',
+            'password-salt': '1234',
+
+            'structural-threshold': '40',  # TODO
+            'temp-dir-path': '/tmp',
+            'docker-mount-base-dir': str(docker_mount_base_dir),
+            'variety-path': 'bin/variety.js',
+         },
+        'database': {
+            'ajax-stats-reload-time': '10000',  # TODO
+            'number-of-latest-firmwares-to-display': '10',
+            'results-per-page': '10'
+        },
+        'default-plugins': {
+            'default': '',
+            'minimal': '',
+        },
+        'expert-settings': {
+            'authentication': 'false',
+            'block-delay': '0.1',
+            'communication-timeout': '60',
+            'intercom-poll-delay': '0.5',
+            'nginx': 'false',
+            'radare2-host': 'localhost',
+            'ssdeep-ignore': '1',
+            'throw-exceptions': 'false',
+            'unpack-threshold': '0.8',
+            'unpack_throttle_limit': '50'
+        },
+        'logging': {
+            'logfile': '/tmp/fact_main.log',
+            'loglevel': 'WARNING',
+        },
+        'unpack': {
+            'max-depth': '10',
+            'memory-limit': '2048',
+            'threads': '4',
+            'whitelist': ''
+        },
+        'statistics': {
+            'max_elements_per_chart': '10'
+        },
+    }
+    # Update recursively
+    for section_name in defaults if defaults else {}:
+        sections.setdefault(section_name, {}).update(defaults[section_name])
+
+    configparser_cfg = ConfigParser()
+    configparser_cfg.read_dict(sections)
+
+    config._parse_dict(sections)
+    cfg = Config(**sections)
+
+    return cfg, configparser_cfg
+
+
+@pytest.fixture
+def cfg_tuple(request):
+    """Returns a `config.Config` and a `configparser.ConfigParser` with testing defaults.
+    Defaults can be overwritten with the `cfg_defaults` pytest mark.
+    """
+    # TODO Use iter_markers to be able to overwrite the config.
```

Comment:
This is a future TODO that I'll probably implement once needed.
The usecase is that one could theoretically have testing classes that inherit some other class which already sets this marker. When the child class also sets this marker both markers should be merged.

I'm not quite sure if we will (and should) have inheritance in testing classes that use this module. So maybe this TODO can be removed in the future.

Reply:
Maybe we can cross this bridge when we get to it :sweat_smile: